gems/aws-sdk-core/lib/aws-sdk-core/xml/builder.rb

```diff
@@ -121,19 +122,27 @@ def timestamp(ref, value)
       # Pass a block if you want to nest XML nodes inside.  When doing this,
       # you may *not* pass a value to the `args` list.
       #
-      def node(name, ref, *args, &block)
+      def node(name, ref, *args, top_level: false, &block)
         attrs = args.last.is_a?(Hash) ? args.pop : {}
-        attrs = shape_attrs(ref).merge(attrs)
+        attrs = shape_attrs(ref, top_level: top_level).merge(attrs)
         args << attrs
         @builder.node(name, *args, &block)
       end
 
-      def shape_attrs(ref)
-        if xmlns = ref['xmlNamespace']
-          if prefix = xmlns['prefix']
-            { 'xmlns:' + prefix => xmlns['uri'] }
-          else
-            { 'xmlns' => xmlns['uri'] }
+      def shape_attrs(ref, top_level: false)
+        # for non top level shapes, do NOT fall back to metadata key on shape
+        xmlns = top_level ? ref['xmlNamespace'] : ref.metadata['xmlNamespace']
```

Comment:
Ignore reviewing anything relating to `xmlNamespace` since I'm waiting on some clarification from the Smithy so the behavior here may change

Reply:
What exactly is the question? Maybe we can figure it out

Reply:
Discussing outside of PR

Reply:
For transparency, I received some behavior clarification on `xmlNamespace` from Kevin. Basically - we are walking back all the changes we made here since this specific protocol-test here need to be fixed.